sprint_9-solution-http-api

MrTflops opened the pull request:
Добавлен пакет http

Содержит всю логику работы HTTP-сервера и обработчиков запросов.

Реализован HttpTaskServer

Запускает сервер на порту 8080.

Поддерживает эндпоинты:

/tasks — работа с задачами

/subtasks — работа с подзадачами

/epics — работа с эпиками

/history — история просмотров

/prioritized — список задач по приоритету

Добавлены обработчики (handlers)

TasksHandler, SubtasksHandler, EpicsHandler, HistoryHandler, PrioritizedHandler

Наследуются от BaseHttpHandler (общая логика отправки/чтения ответов).

Поддержка JSON (Gson)

Сериализация/десериализация задач.

Адаптеры для LocalDateTime и Duration.

Тесты

Интеграционные тесты (HttpTaskServerTest).

Модульные тесты для обработчиков (TasksHandlerTest и др.).

Обновлен TaskManager

Добавлены методы для работы с приоритетами (getPrioritizedTasks).

Проверка пересечений задач по времени.

MrTflops committed:
====================
InMemoryHistoryManager: реализован двусвязный список
- добавлен Node для узлов двусвязного списка
- реализованы методы linkLast() и removeNode() для работы со списком
- использована HashMap для быстрого поиска задач
- интегрирована новая история просмотров в TaskManager
- добавлены тесты для проверки функциональности
====================
TaskManager: обновил для работы с новой историей
- заменен простой список history на InMemoryHistoryManager
- обновлены методы getTask(), getEpic(), getSubtask()
- добавлено удаление задач из истории при вызове deleteTaskById()
- вроде как обеспечил согласованность истории при удалении задач
====================
test: добавленны тесты для HistoryManager
- тесты на добавление/получение задач из истории
- проверка граничных случаев (пустая история, дубликаты)
- тесты на удаление из разных позиций списка
- проверка целостности истории после удалений
====================

MrTflops committed:
test

MrTflops committed:
внес правки по checkstyle

MrTflops committed:
внес правки по checkstyle

MrTflops committed:
внес правки в Epic, Subtask, Task для прохождения теста

MrTflops committed:
добавил тестов

MrTflops committed:
рафакторинг кода

MrTflops committed:
Sprint 6

MrTflops committed:
Sprint 6

MrTflops committed:
Sprint 6

MrTflops committed:
Sprint 6/добавил рефлексию

MrTflops committed:
Sprint 6

MrTflops committed:
Sprint 6

MrTflops committed:
Sprint 6

MrTflops committed:
Sprint 6

MrTflops committed:
Sprint 6

MrTflops committed:
Sprint 6

MrTflops committed:
Sprint 6

MrTflops committed:
Sprint 6

MrTflops committed:
Sprint 6

MrTflops committed:
Sprint 6

MrTflops committed:
Sprint 6

MrTflops committed:
Sprint 6

MrTflops committed:
Sprint 6

MrTflops committed:
Sprint 6

Commit:
sprint-6-solution

Commit:
nemoryaskanagerest - изменил имена у методов, nemoryistoryanagerest - изменил имена у методов

Commit:
- Создан enum Type для типов задач (TASK, EPIC, SUBTASK)
- реализованы методы сериализации:
  * taskToString(Task) - преобразование задачи в CSV-строку
  * taskFromString(String) - восстановление задачи из строки
- обавлено автосохранение после каждой модификации данных
- реализована загрузка состояния из файла (loadFromFile)
- улучшена обработка ошибок (ManagerSaveException)
- обновлена валидация данных при загрузке
- оптимизировано чтение/запись файлов

Commit:
правки в error: invalid flag: ./test/service/FileBackedTaskManagerTest

Commit:
1

Commit:
1

Commit:
1

Commit:
1

Commit:
1

Commit:
1

MrTflops committed:
7

MrTflops committed:
добавил файл tests

MrTflops committed:
исправил ошибку в названии файла

MrTflops committed:
правка ошибки в файле tests

MrTflops committed:
7

MrTflops committed:
1

MrTflops committed:
1

MrTflops committed:
в методе taskFromString используется теперь String.format

MrTflops committed:
корректировка названий методов

Commit:
sprint 9

Commit:
sprint 9

Commit:
sprint 9

Commit:
sprint 9

Commit:
sprint 9

Commit:
sprint 9

Commit:
поправил ошибку в объявлении prioritizedTasks

Commit:
9

Commit:
9

Commit:
9

MrTflops committed:
9

MrTflops committed:
9

MrTflops committed:
9

MrTflops committed:
9

MrTflops committed:
9

MrTflops committed:
9

MrTflops committed:
9

MrTflops committed:
9

MrTflops committed:
9

felixoff commented on src/main/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,162 @@
+                }
+            }
+        } catch (IOException e) {
+            throw new ManagerSaveException("Ошибка при загрузке из файла", e);
```
Скорее при вводе выводе

MrTflops replied:
в методе loadFromFile - скорректировал текст ошибки + добавил String.format

Commit:
10/04/16-56

Commit:
10/04/14:01 добавил импорты в хттптасксервер

Commit:
10/04/15:50

Commit:
10/04/15:50

Commit:
10/04/15:50

Commit:
10/04/ убран лишний импорт LocalDateTime

Commit:
10/04/ чекстайл taskmanager

Commit:
10/04/ чекстайл taskmanager

Commit:
10/04/16^52

Commit:
10/04/16^52

Commit:
новый класс NotFoundException Добавил недостающие методы в интерфейс TaskManager, реализовал новые методы в InMemoryTaskManager, добавил методы sendResponse и readRequestBody в BaseHttpHandler,
исправил конструктор FileBackedTaskManager, обновил импорты в EpicsHandler

Commit:
чекстайл

Commit:
чекстайл

Commit:
изменили модификаторы доступа методов updateEpicStatus и updateEpicTimes в InMemoryTaskManager с private на protected

Commit:
Nсправлены методы работы с задачами (getAllTasks вместо getTasks, исправлена работа с приоритетными задачами (List вместо Set), исправлена работа с временем
 (правильное использование Duration и LocalDateTime), добавлены несуществующие методы (deleteTaskById, isTaskOverlapping), исправлены обращения к gson через HttpTaskServer.getGson()

Commit:
test

MrTflops committed:
home

MrTflops committed:
home

MrTflops committed:
home

MrTflops committed:
home

MrTflops committed:
home

MrTflops committed:
home

MrTflops committed:
тесты TasksHandlerTest

MrTflops committed:
тесты TasksHandlerTest

MrTflops committed:
тесты TasksHandlerTest

MrTflops committed:
тесты FileBackedTaskManagerTest

MrTflops committed:
уже близок к тому чтобы выбросить комп в окно

MrTflops committed:
уже близок к тому чтобы выбросить комп в окно

MrTflops committed:
уже близок к тому чтобы выбросить комп в окно

MrTflops committed:
уже близок к тому чтобы выбросить комп в окно

MrTflops committed:
уже близок к тому чтобы выбросить комп в окно

MrTflops committed:
час ночи

Commit:
TaskManagerTest, FileBackedTaskManagerTest

Commit:
TaskManagerTest, FileBackedTaskManagerTest

Commit:
InMemoryTaskManager

Commit:
InMemoryTaskManager

Commit:
правки в пакет model

Commit:
правки в пакет model_V_2

Commit:
правки в пакет model

MrTflops committed:
очередная правка в тестах

MrTflops committed:
очередная правка в тестах

MrTflops committed:
очередная правка в тестах

MrTflops committed:
11/04

MrTflops committed:
9

MrTflops committed:
Merge remote-tracking branch 'origin/sprint_9-solution-http-api_V_2' into sprint_9-solution-http-api_V_2

MrTflops committed:
subbota

MrTflops committed:
vsk

MrTflops committed:
vsk

MrTflops committed:
что-то правлю не понимая что. в надежде что все попраится

MrTflops committed:
исправил BaseHttpHandler

MrTflops committed:
исправил BaseHttpHandler

MrTflops committed:
исправил тасксервертест
